### PR TITLE
Refactor mysql, mongo, redis interface

### PR DESCRIPTION
**Pull request by mialinx**
Use WALG_STREAM_CREATE_COMMAND / WALG_STREAM_RESTORE_COMMAND
instead of piping Stdin/Stdout

### CODE REVIEW

File: internal/databases/mongo/backup_push_handler.go [hunk 1]
```diff
@@ -3,16 +3,19 @@ package mongo
...
 )
 
 // HandleStreamPush starts backup procedure.
-func HandleStreamPush(uploader archive.Uploader, command []string, metaProvider archive.BackupMetaProvider) {
+func HandleStreamPush(uploader archive.Uploader, metaProvider archive.BackupMetaProvider) {
 	err := metaProvider.Init()
 	tracelog.ErrorLogger.FatalOnError(err)
-	waitAndFatalOnError, stream := internal.StartCommand(command)
-	err = uploader.UploadBackup(stream, metaProvider)
+	cmd, err := internal.GetCommandSetting(internal.NameStreamCreateCmd)
```
[the remark this answers is not in the document]
Comment (mialinx) [reply internal/databases/mongo/backup_push_handler.go]: From my point of view, cmd/*  packages should do nothing but dealing with commandline arguments. All app initialization code should be placed in internal package.

Ok, let's move it to cmd/mongo package, may be will refactor it later for all databases;

File: internal/configure.go [hunk 2]
```diff
@@ -308,39 +309,28 @@ func GetSentinelUserData() interface{} {
...
-	if err != nil {
-		return nil, err
+	shell := os.Getenv("SHELL")
+	if shell == "" {
+		shell = "/bin/sh"
 	}
-	return val, nil
+	cmd := exec.CommandContext(ctx, shell, "-c", dataStr)
+	// do not shut up subcommands by default
+	cmd.Stderr = os.Stderr
```
[the remark this answers is not in the document]
Comment (mialinx) [reply internal/configure.go]: It's default behaviour for shell commands, so i left it as is.
I've created StartCommandWithStdoutStderr util, that do all that stuff.

File: internal/configure.go [hunk 3]
```diff
@@ -308,39 +309,28 @@ func GetSentinelUserData() interface{} {
 	return out
 }
 
-func getCommandFromEnvAndParse(variableName string) ([]string, error) {
+func GetCommandSettingContext(ctx context.Context, variableName string) (*exec.Cmd, error) {
```
[the remark this answers is not in the document]
Comment (mialinx) [reply internal/configure.go]: For now exec.Cmd are always created from config value.
I think there is no need in separated exec.Cmd creator at the moment.

[the remark this answers is not in the document]
Comment (perekalov) [reply internal/configure.go]: We have mixed two separate jobs here. Imao it breaks single responsibility principle

[the remark this answers is not in the document]
Comment (perekalov) [reply internal/configure.go]: It's not major issue btw

[the remark this answers is not in the document]
Comment (mialinx) [reply internal/configure.go]: Think of it like  getDatetimeSetting, getDurationSetting or any other config getter returning value in a convenient format. Just a shortcut.

**Pull request merged**